src/Data/JSONPath/Execute.hs

```diff
@@ -64,13 +64,22 @@ executeJSONPathElement s@(Search js) val =
      else ResultList $ x ++ y
 
 valMap :: ToJSON b => (Value -> ExecutionResult b) -> Value -> [ExecutionResult b]
-valMap f v@(Object o) = map snd . Map.toList $ Map.map f o
+valMap f (Object o) = map snd . Map.toList $ Map.map f o
 valMap f (Array a) = V.toList $ V.map f a
 valMap _ v = pure $ ResultError $ "Expected object or array, found " <> (encodeJSONToString v)
 
 executeCondition :: Value -> Condition -> Literal -> Bool
-executeCondition (Number n1) Equal (LitNumber n2) = n1 == (fromInteger $ toInteger n2)
+executeCondition val NotEqual lit = not $ executeCondition val Equal lit
+executeCondition (Number n1) Equal (LitNumber n2) = n1 == realToFrac n2
 executeCondition (String s1) Equal (LitString s2) = s1 == s2
+executeCondition _ Equal _ = False
+executeCondition val GreaterThan lit =
+  not (executeCondition val SmallerThan lit) &&  not (executeCondition val Equal lit)
```

Comment:
:information_source: The idea here is to implement all operators in terms of `==` and `<`